src/shell.ts

```diff
@@ -69,8 +69,11 @@ export class Shell {
     this.shell.stdin!.write(s);
   }
 
+  public async run(script: string): Promise<Output<Buffer> & (Exit | Signal)>;
+  public async run(script: string, encoding: string): Promise<Output<string> & (Exit | Signal)>;
+
   // Passes script into shell, returns output and error
-  public async run(script: string): Promise<Output & (Exit | Signal)> {
+  public async run(script: string, encoding?: string): Promise<Output<any> & (Exit | Signal)> {
```

Comment:
This type is at least `Buffer | string`, not `any`.  But isn't this an overload?  If you don't pass `encoding` it's a `Buffer`, otherwise it's a `string`.  So overload on `encoding` and return a more exact type, please.  It's more work here, and _much_ safer code when using.